lib/services/amazon_sns.rb

```diff
@@ -35,17 +42,43 @@ def publish_to_sns(cfg, json)
         :secret_access_key => cfg['aws_secret']
       })
 
-      sns.publish({
-        :message            => json,
-        :topic_arn          => cfg['sns_topic'],
-        :message_attributes => message_attributes
-      })
-    rescue Aws::SNS::Errors::AuthorizationErrorException => e
-      raise_config_error e.message
-    rescue Aws::SNS::Errors::NotFoundException => e
-      raise_missing_error e.message
-    rescue SocketError
-      raise_missing_error
+      if json.bytesize <= MAX_MESSAGE_SIZE
+        sns.publish({
+          :message            => json,
+          :topic_arn          => cfg['sns_topic'],
+          :message_attributes => message_attributes
+        })
+      end
```

Comment:
Slightly confused. If the message is less than the maximum size, shouldn't we just send that *one* message and be done? The `if end` here will still try to do the second set of messages below.

We also want to make sure the `rescue`s below work for either. I may be worth building a helper method that does the `sns.publish` to clean up the code a fair bit.